### ISSUE TITLE
Community screen issues (follow up of #14018)

### ISSUE THREAD

**Issue by pavloburykh** [status-im/status-react#14291]
This ticket is a follow up of [update community home screens](https://github.com/status-im/status-mobile/pull/14018#top)
#14018

### ISSUE 1 Scrolling on initial community screen doesn't work

**Steps:** 

1. Add a lot of communities (more that list can display)
2. Try to scroll through the list
3. Observe the result

**Actual result:** scrolling doesn't work

**Expected result:** scrolling works

https://user-images.githubusercontent.com/97245802/198568030-309fcd79-91d6-42ca-8858-5728d044aa74.mp4

### ISSUE 2 Empty community screen misses some elements that Figma design has: icons at Discover card, text and image in the empty list

**Actual:**
![photo_2022-10-28 13 44 41](https://user-images.githubusercontent.com/97245802/198569539-7027ae40-553a-4e5d-b55b-9d050f9f4ac5.jpeg)

**Expected:** 

![Communities for Mobile – Figma 2022-10-28 13-43-52](https://user-images.githubusercontent.com/97245802/198569574-d0080e3e-783a-4b2b-9923-d7e276bd6a23.png)

### ISSUE 3 Missing accessability ids for some elements

1. Discover card

<img width="1242" alt="Appium 2022-10-28 14-09-19" src="https://user-images.githubusercontent.com/97245802/198573706-d5bef1db-b6c7-452f-87a0-10857282df8f.png">

2. Filter buttons (Joined, Pending, Opened) 

<img width="427" alt="Appium 2022-10-28 14-10-55" src="https://user-images.githubusercontent.com/97245802/198573895-9b51cabc-71b0-4933-bb11-c74d24e6dce9.png">

3. Discover Communities scrollbar with categories elements

<img width="420" alt="Appium 2022-10-28 16-27-15" src="https://user-images.githubusercontent.com/97245802/198610348-3e1cc0a2-2d39-4fd0-9d20-5c424af9e16e.png">

4. Filter tabs, Sort and View buttons on Discover Communities screen

<img width="423" alt="Appium 2022-10-28 16-29-49" src="https://user-images.githubusercontent.com/97245802/198612465-02896813-f249-4dc7-8987-864b59498a7f.png">

5. Sort Communities options elements

<img width="427" alt="Appium 2022-10-28 16-37-28" src="https://user-images.githubusercontent.com/97245802/198616637-b487c507-07b9-48f7-84c1-e83cbac77600.png">

### ISSUE 4 White screen splashes when redirecting to Discover communities screen (might be some global problem, not PR related)

I am reproducing an ISSUE with 4+ communities added to the list. Better to reproduce with the dark mode on. 

**Steps:** 

1. Add 4+ communities 
2. Open Discover communities screen
3. Pay attention at white screen that flashes during redirection

**Actual result:**
White screen splashes when redirecting to Discover communities screen

**Expected result:** no white screen splash

https://user-images.githubusercontent.com/97245802/198599715-71bc9c0b-a970-4067-be5d-2e987dc4a582.mp4

### ISSUE 5 Sort Communities options are missing icons

**Actual:**

![photo_2022-10-28 16 20 09](https://user-images.githubusercontent.com/97245802/198605537-147c3ea7-3c29-4282-8798-f3d4cb0c5e9e.jpeg)

**Figma:**

![Communities for Mobile – Figma 2022-10-28 16-21-44](https://user-images.githubusercontent.com/97245802/198606618-7fbc8280-bae9-44f4-94f4-1071eff9eeaa.png)

### ISSUE 6 Expand categories is missing on Discover communities screen

**Figma:** 

![Communities for Mobile – Figma 2022-10-28 16-41-47](https://user-images.githubusercontent.com/97245802/198620002-bd26526b-eab0-4a91-bfc6-e386ddb00c3d.png)

**Comment by jo-mut** [comment]
hi @pavloburykh thank you for testing my pr #14018 

I am aware that some of the issue you have found with the PR are not implemented as is on the designs. The intention with the current implementation on the #14018 is to address some on the issues on the community screen while still keeping the PR small to be well reviewed.

Following our discusion with the design team, some of the things addressed in the PR have been scoped out. Just to mention some of those captured in this issue

**ISSUE 5 & ISSUE 6**

These two issues have been scroped out and are to be implemented on post MVP.

**ISSUE 4**

The white screen splashes is not specific to this PR. Its caused by our implementation of navigation and is expected to happen. I wouldnt say its a bug at this point. cc @Parveshdhull 


**ISSUE 2**

The discover card images are placeholders. The final implementation would be probably be different. Since using images would mean that they will be included in the release we decided to used react native view component instead of images to achieve the same design layout as images are not that much necessary in this case.

As for the communities empty state placeholder  case, its not that much necessary to add the design as this is just at the moment an illustration and is not to be implemented by the dev team as i understand from @pedro-et  

**THING OF CONCERN**

**ISSUE 1**

I would prefer to add communities scroll behaviours in a separate PR just so that I can avoid #14018 becoming such a large PR. The scrolling behaviours would include include global scroll behaviours which would make the PR large.

**ISSUE 2**

I will add the accessabilty id as you have highlighted.

Thank you @pavloburykh

**Comment by Parveshdhull** [comment]
hi @jo-mut Thank you for pinging me. I can take a look into Issue 4 once the PR is merged.

**Comment by pavloburykh** [comment]
> #14018

Thank @jo-mut ! I've created this separate issue in order not to block https://github.com/status-im/status-mobile/pull/14018
So I guess, you can make a separate PR and link it to this issue with fixes of ISSUE 1 and ISSUE 2. Thanx!

**Comment by Parveshdhull** [comment]
> hi @jo-mut Thank you for pinging me. I can take a look into Issue 4 once the PR is merged.

Hi @jo-mut  I checked issue 4 and it doesn't look related to navigation.
As navigation is working fine with less number of communities.
It is most probably a performance issue, which caused this delay and white screen (on some devices and black on others) with 4+ communities. I guess we need to optimize these community cards.

**Comment by Parveshdhull** [comment]
maybe @flexsurfer might have some tips/ideas about this behavior, he is expert on performance :)
